test/linking_tests.bzl

```diff
@@ -42,76 +42,92 @@ def linking_test_suite(name):
     Args:
         name: The name to be included in test names and tags.
     """
-    default_test(
-        name = "{}_default_apple_link_test".format(name),
-        tags = [name],
-        expected_argv = [
-            "-Xlinker",
-            "-objc_abi_version",
-            "-Xlinker",
-            "2",
-            "-ObjC",
-        ],
-        not_expected_argv = [
-            "-g",
-            "DSYM_HINT_LINKED_BINARY",
-            "-dead_strip",
-        ],
-        mnemonic = "ObjcLink",
-        target_under_test = "//test/test_data:macos_binary",
-    )
+    framework_flags = {
+        "macos": ["-framework", "Foundation"],
+        "ios": ["-framework", "Foundation", "-framework", "UIKit"],
+    }
 
-    opt_test(
-        name = "{}_opt_link_test".format(name),
-        tags = [name],
-        expected_argv = [
-            "-Xlinker",
-            "-objc_abi_version",
-            "-Xlinker",
-            "2",
-            "-ObjC",
-            "-dead_strip",
-        ],
-        mnemonic = "ObjcLink",
-        target_under_test = "//test/test_data:macos_binary",
-    )
+    for platform in ["macos", "ios"]:
```

Comment:
i think this is probably overkill, you can probably just add the framework to the default case, and then create a new rule at the top of the file to transition the features off, and assert that without any new targets and without adding it to each test

Reply:
I'd be happy to do that, do you know of any good examples of how to transition on a feature? This is an area of Bazel I have not yet developed muscles for yet.

Reply:
https://github.com/bazelbuild/apple_support/blob/288cfc0646781abab6ce4372e34d335ad9245768/test/linking_tests.bzl#L19-L21

i just mean with this test infra

Reply:
that makes sense, thank you!